app/javascript/components/App.js

```diff
@@ -38,24 +39,35 @@ readSchool = () => {
 }
 
   render () {
+    const {
+      current_user
+    } = this.props
+
     return (
       <>
-
         <Router>
         <Header {...this.props}/>
           <Switch>
             <Route exact path="/" component={HomePage} />
             <Route path="/schoolindex" render={(props) => <SchoolIndexPage schools={this.state.schools}/>} />
             <Route path="/reviewindex" render={(props) => <ReviewIndexPage reviews={this.state.reviews}/>} />
-          
-          <Route path="/schoolshow/:id" render ={(props) => {
-            let id = props.match.params.id
-            let school = this.state.schools.find(school => school.id === +id)
-            return <ShowSchoolPage school={school}/> }} />
-          <Route path="/reviewshow/:id" render ={(props) => {
-            let id = props.match.params.id
-            let review = this.state.reviews.find(review => review.id === +id)
-            return <ShowReviewPage review={review}/> }} />
+            
+            <Route path="/myreviews" render={(props) => {
+              if(current_user) {
+                let myReviews = this.state.reviews.filter(review => review.user_id === current_user.id)
+                return <ProtectedReviewIndexPage review={myReviews} schools={this.state.schools}/>
+              } else {
+                return <Route path="/reviewindex" render={(props) => <ReviewIndexPage reviews={this.state.reviews} schools={this.state.schools}/>} />
+              }
```

Comment:
This could be misleading to your user if they click on myreviews and are routed to review index.